client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -166,91 +165,56 @@ const currentLineWithoutOutputVariableName = (
     if (errorWasThrown) return;
-    let currentLine = line.trim();
-    const currentLineDefinesNewApplication = declaredApplications.includes(
-      currentLine
-    );
-    const currentLineContainsElementName = regexForElementNameLine.test(
-      currentLine
-    );
-    const currentLineIncludesSplitPlaceholder = currentLine.includes(
-      splitPlaceholder
-    );
-    const currentLineHasNoSpecifiedApplication =
-      typeof currentApplication === 'undefined';
-    const currentLineDefinesOutputValue = regexForOutputVariable.test(
-      currentLine
-    );
+    let currentLine = line;
+    const currentLineIncludesSplitPlaceholder = currentLine.includes(splitPlaceholder);
+    const currentLineDefinesOutputValue = regexForOutputVariable.test(currentLine);
+    const currentLineStartsWithFourspace = currentLine.startsWith(FOURSPACE);
 
-    if (currentLineDefinesNewApplication) {
-      currentApplication = currentLine;
-      return;
-    }
-    if (currentLineHasNoSpecifiedApplication) {
-      customNotification(
-        'Error',
-        `There is no RPA-Application specified for line "${currentLine}"`
-      );
-      errorWasThrown = true;
+    if (!currentLineStartsWithFourspace) {
+      instructionBlocks.push({ name: currentLine });
       return;
     }
 
     if (currentLineIncludesSplitPlaceholder) {
-      customNotification(
-        'Error',
-        `It is not allowed to use & or § as param values \nError location: "${line}"`
-      );
+      customNotification('Error', `It is not allowed to use & or § as param values \nError location: "${line}"`);
       errorWasThrown = true;
       return;
     }
 
-    if (currentLineContainsElementName) {
-      instructionBlocks.push({
-        rpaApplication: currentApplication,
-        name: currentLine.substring(1).trim(),
-      });
-      return;
-    }
-
-    currentLine = currentLine.replace(/( {4})/g, splitPlaceholder);
+    currentLine = currentLine.trim().replace(/( {4})/g, splitPlaceholder);
 
     if (currentLineDefinesOutputValue) {
       const outputValueName = getOutputName(currentLine);
       instructionBlocks[
         instructionBlocks.length - 1
       ].outputName = outputValueName;
 
-      currentLine = currentLineWithoutOutputVariableName(
-        currentLine,
-        splitPlaceholder
-      );
+      currentLine = currentLineWithoutOutputVariableName(currentLine, splitPlaceholder);
     }
 
     if (!errorWasThrown) {
-      instructionBlocks[instructionBlocks.length - 1].rpaTask = getRpaTask(
-        currentLine,
-        splitPlaceholder
-      );
-      instructionBlocks[
-        instructionBlocks.length - 1
-      ].paramArray = getRpaParameters(
-        currentLine,
-        splitPlaceholder,
-        instructionBlocks
-      );
+      const rpaTask = getRpaTask(currentLine, splitPlaceholder);
+      const matchingCombination = taskAndApplicationCombinations.filter((singleCombination) => singleCombination.Task === rpaTask)[0]
```

Comment:
Have you found cases, when there are multiple in here? Shouldnt this be handled? See above Outlook vs Word

Reply:
Won't this also throw an error, when you try to access it and the array is empty? So the following error handling might be useless, isn't it?

Reply:
Should be fixed now